### PR TITLE
Adds Params handler specs

### PR DESCRIPTION
**Pull request by eliasjpr**
This is necessary to have tests against existing functionality.

### CODE REVIEW

File: spec/amber/pipe/params_spec.cr [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+        params.call(context)
+
+        context.params["test"].should eq "test"
+      end
+
+      it "parses body params with charset" do
+        headers = HTTP::Headers.new
+        headers["Content-Type"] = "application/x-www-form-urlencoded; charset=UTF-8"
+        body = "test=test"
```
Comment (bew) [inline spec/amber/pipe/params_spec.cr]: put some utf-8 chars here?

Comment (eliasjpr) [inline spec/amber/pipe/params_spec.cr]: Can you give me more context about the utf-8 chars? Do you mean something like test=\x3D?

File: spec/amber/pipe/params_spec.cr [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        context = create_context(request)
+        params = Params.instance
+
+        params.call(context)
+
+        context.params["test"].should eq "test"
+      end
+
+      it "parses json array" do
+        headers = HTTP::Headers.new
```
Comment (bew) [inline spec/amber/pipe/params_spec.cr]: note: would be great to be able to do sth like:
```cr
headers = HTTP::Headers{"Content-Type": "application/json"}
```

File: spec/amber/pipe/params_spec.cr (hunk 1 above)
Comment (bew) [inline spec/amber/pipe/params_spec.cr]: Yes! I mean some text that's not ascii (ex: `¿øœ`)

**Pull request merged**